arch/arm/core/cortex_m/Kconfig

```diff
@@ -106,16 +106,31 @@ config ISA_THUMB2
 config CPU_CORTEX_M_HAS_BASEPRI
 	bool
 	# Omit prompt to signify "hidden" option
+	depends on ARMV7_M_ARMV8_M_MAINLINE
 	default n
 	help
 	  This option signifies the CPU has the BASEPRI register.
 
+	  The BASEPRI register defines the minimum priority for
+	  exception processing. When BASEPRI is set to a nonzero
+	  value, it prevents the activation of all exceptions with
+	  the same or lower priority level as the BASEPRI value.
+	  Always present in CPUs that implement the ARMv7-M or
+	  ARM8-M Mainline architectures.
+
 config CPU_CORTEX_M_HAS_VTOR
 	bool
 	# Omit prompt to signify "hidden" option
+	depends on !CPU_CORTEX_M0
```

Comment:
I don't think this is correct, I think there are cases of M0's having VTOR.  Look at arch/arm/soc/st_stm32/stm32l0/Kconfig.series